views/ddox.docpage.dt

```diff
@@ -14,60 +16,79 @@ block ddox.title
 		- title = prefix ~ names.join("/") ~ " - multiple declarations";
 	
 block ddox.members
+	- bool multi_page = info.docGroups.length > 1;
 
-	- if (info.docGroups.length > 1)
-		ul
-			- foreach (dg; info.docGroups)
-				- auto itm = dg.members[0];
-				li #{itm.kindCaption} #{itm.nestedName}
+	- if (multi_page)
+		nav.page-nav
+			ul
+				- foreach (i, dg; info.docGroups)
+					- auto itm = dg.members[0];
+					li: a(href="\##{i}") #{dg.kinds.joiner("/")} #{itm.nestedName}
 
+	include ddox.inc.utils
 	include ddox.inc.composite
 	include ddox.inc.enum
 	include ddox.inc.function
 	include ddox.inc.template
 	include ddox.inc.variable
 
-	- foreach (dg; info.docGroups)
-		- auto item = cast(Declaration)dg.members[0];
-		- assert (item !is null, "Unknown entity type: "~dg.members[0].classinfo.toString());
+	- foreach (i, dg; info.docGroups)
+		- bool got_function = false;
+		- int hlevel = 2 + multi_page;
+		- auto kinds = dg.kinds;
 
-		section
-			- if (info.docGroups.length > 1)
-				h2 #{item.kindCaption} #{item.nestedName}
+		- if (multi_page)
+			h2(id=i) #{kinds.joiner("/")} #{dg.members[0].nestedName}
 
-			p!= info.formatDoc(dg, 3, sec => sec == "$Short")
-			|!= info.formatDoc(dg, 3, sec => sec == "$Long")
+		p!= info.formatDoc(dg, hlevel, sec => sec == "$Short")
+		|!= info.formatDoc(dg, hlevel, sec => sec == "$Long")
+
+		- foreach (itement; dg.members)
+			- auto item = cast(Declaration)itement;
+			- assert (item !is null, "Unknown entity type: "~dg.members[0].classinfo.toString());
+			- auto shlevel = hlevel;
+
+			- if (got_function && cast(FunctionDeclaration)item)
+				- continue;
+
+			- if (kinds.walkLength > 1)
+				- shlevel++;
+				- if (hlevel == 2)
+					h2 #{item.kindCaption} #{item.nestedName}
+				- else
+					h3 #{item.kindCaption} #{item.nestedName}
```

Comment:
Forgot to change this when I introduced `heading` later in the process. Will fix.